---
title: 'iSEE workshop - package and functionality'
subtitle: 'Edition 2020'
author: '[Charlotte Soneson](https://csoneson.github.io), [Kevin Rue-Albrecht](https://kevinrue.github.io), [Federico Marini](https://federicomarini.github.io/)'
institute: ''
date: '2020-06-08 (updated: `r Sys.Date()`)'
output:
  xaringan::moon_reader:
    css: [default, metropolis, rladies-fonts, "my-theme.css"]
    lib_dir: libs
    nature:
      highlightStyle: github
      highlightLines: true
      countIncrementalSlides: false
# uncomment this line to produce HTML and PDF in RStudio:
# knit: pagedown::chrome_print
---

```{r, load_refs, include=FALSE, cache=FALSE}
options(htmltools.dir.version = FALSE)
library(RefManageR)
BibOptions(
  check.entries = FALSE,
  bib.style = "authoryear",
  cite.style = "authoryear",
  max.names = 2,
  style = "markdown",
  hyperlink = "to.doc",
  dashed = TRUE)
bib <- ReadBib("references.bib")
```

layout: true

<div class="my-header"><img src="https://raw.githubusercontent.com/Bioconductor/BiocStickers/master/iSEE/iSEE.png" alt="OxfordiSEE logo" align="right" height="90%"></div>

<div class="my-footer"><span>
iSEE Workshop 2020
&emsp;&emsp;&emsp;&emsp;&emsp;&emsp;&emsp;&emsp;
interactive SummarizedExperiment Explorer
</span></div>

---

# The team

<br/>

<table>
<tr>
<td width="25%"><a href="https://kevinrue.github.io"><img src="img/kevin-rue.jpg"></a></td>
<td width="25%"><a href="https://csoneson.github.io"><img src="img/charlotte-soneson.png"></a></td>
<td width="25%"><a href="https://federicomarini.github.io"><img src="img/federico-marini.png"></a></td>
<td width="25%"><a href="https://orcid.org/0000-0002-3564-4813"><img src="img/aaron-lun.png"></a></td>
</tr>
<tr>
<td align='center' width="25%">Kevin</td>
<td align='center' width="25%">Charlotte</td>
<td align='center' width="25%">Federico</td>
<td align='center' width="25%">Aaron</td>
</tr>
</table>

???

This was a team effort!

---

# Designed for Bioconductor

```{r, fig.align='center', out.width='75%',  echo=FALSE}
knitr::include_graphics("img/biocstickers.jpg")
```

<https://bioconductor.org>

???

iSEE tightly integrates with other packages of the Bioconductor project.

---

# Original wishlist (1 / 2)

- An interactive user interface for exploring data in `r BiocStyle::Biocpkg("SummarizedExperiment")`

- Particular focus given to single-cell data in the `r BiocStyle::Biocpkg("SingleCellExperiment")` derived class

- Sample-oriented visualizations (e.g., patients, cells)

- Feature-oriented visualizations (e.g., genes, protein, metabolites, loci, variants)

- Heatmaps (cells _and_ features)

- Selectable points

???

When we sat down at EuroBioc2017, we drafted a wishlist of features.
See commit #3 at https://github.com/csoneson/iSEE/tree/021e3e20cfdb194e511f8097ed544329bd46bcd6

---

# Original wishlist (2 / 2)

- Colouring of samples by metadata or assayed values

- Stratify axes and facets by metadata (e.g., violin plots)

- Hover and click

- Zoom

- Transmission of points selections between plots

- Code tracking for reproducibility and batch generation of figures

???

When we sat down at EuroBioc2017, we drafted a wishlist of features.
See commit #3 at https://github.com/csoneson/iSEE/tree/021e3e20cfdb194e511f8097ed544329bd46bcd6

---

# SingleCellExperiment

```{r, fig.align='center', out.height='400px', echo=FALSE}
knitr::include_graphics("img/OSCA/bioc-figures_v2-02.png")
```

`r Citet(bib, "amezquita2020osca")`

???

iSEE focuses on the SingleCellExperiment class.
This class stores all the data and metadata associated with assays, cells, and features.

---

# Usage in a workflow

.pull-left[
- Add

- Points

- Here
]

.pull-right[
```{r, fig.align='center', echo=FALSE}
knitr::include_graphics("img/OSCA/bioc-figures_v2-03.png")
```

.right[
`r Citet(bib, "amezquita2020osca")`
]
]

???

<!-- 
The SingleCellExperiment class is designed to accomodate all the information produced along a typical single-cell analysis workflow.
Those data include raw data:
- raw assay data
- experimental metadata
Processed data:
- quality control metrics
- normalized data
- dimensionality reduction results
Downstream analyses:
- cluster labels
- differential expression results
- downstream cell and feature annotations
-->